nested_lookup/lookup_api.py

```diff
@@ -29,28 +31,134 @@ def _nested_delete(document, key):
     return document
 
 
-def nested_update(document, key, value, in_place=False):
+def nested_update(document:dict, key:str, value:object, in_place:bool=False, treat_list_as_element:bool = False) -> dict:
+    """
+    Method to update a key->value pair in a nested document
+    Args:
+        document: Might be List of Dicts (or) Dict of Lists (or)
+            Dict of List of Dicts etc...
+        key: Key to update the value
+        value: Value to set
+        in_place (bool): 
+            True: modify the dict in place; 
+            False: create a deep copy of the dict and modify it
+            Defaults to False
+        treat_list_as_element (bool):
+            True: if a list is provided as "value", the function trys to match the list elements to the occurences of the key.
+                If the key occures more often than the provided list has elements, the first element gets recycled.
+            False: the provided list is treated as one scalar value and will be set as value to every key that matches.
+            Defaults to False.
+    Return:
+        (dict) Returns a document that has updated key, value pair.
+    """
+
+    # check if a list or scalar value is provided and create a list from the scalar value
+    # check the length of the list and provide it to _nested_update
+    if isinstance(value, list) and not treat_list_as_element:
```

Comment:
We need to alert the user if he fails to pass list as a value when he opt for the feature
```suggestion
    if not treat_list_as_element and not isinstance(value, list):
        raise Exception('You need to pass value as list if you opt for this feature')
    val_len = 1 if treat_list_as_element else len(value)
```

Reply:
Done